### ISSUE TITLE
Is your denoiser directly taken from https://github.com/openai/guided-diffusion 256x256 diffusion (not class conditional): 256x256_diffusion_uncond.pt?

### ISSUE THREAD

**Issue by xiximelon** [yuanzhi-zhu/DiffPIR#21]
![image](https://github.com/yuanzhi-zhu/DiffPIR/assets/72423868/66021a8a-1953-407e-a89c-9f893cc43ab5)

**Comment by yuanzhi-zhu** [comment]
Yes ;)

**Comment by xiximelon** [comment]
thanks!

**Comment by xiximelon** [comment]
您好，不好意思再次打扰，想请问：
1.您训练FFHQ diffusion model的超参数是什么？这写参数是如何决定的呢？
2.训练的iteration如何确定，是靠后面sample的推理结果观察看哪一个.pth的结果更好吗？

**Comment by xiximelon** [comment]
@yuanzhi-zhu 您好，不好意思再次打扰，想请问：
1.您训练FFHQ diffusion model的超参数是什么？这里参数是如何决定的呢？
2.训练的iteration如何确定，是靠之后image_sample.py的推理结果观察看哪一个.pth的结果更好吗？

**Comment by yuanzhi-zhu** [comment]
您好！
Our work proposed a training-free method for image restoration, so we did not really train a model.

Anyway,
1. you may want to check the paper of guided diffusion for the choice of these hyper-parameters
2. you are right, for diffusion models training, it is common to sample the checkpoints for sanity checks and calculate the fid score to help find a good checkpoint.

**Comment by xiximelon** [comment]
Thanks ! I got it !

**Comment by yuanzhi-zhu** [comment]
We choose the minimum iterations in DiffPIR empirically for different IR tasks. 20 is enough for SR and deburring, and 100 is usually required for inpainting to get a better result.
And the iterations used in DiffPIR are independent of those in DPIR.

**Comment by xiximelon** [comment]
> We choose the minimum iterations in DiffPIR empirically for different IR tasks. 20 is enough for SR and deburring, and 100 is usually required for inpainting to get a better result. And the iterations used in DiffPIR are independent of those in DPIR.
 Thanks again !  I  wonder in inpaiting(random) tasks, the mask ratio is set to which number? is it 50%?

<img width="118" alt="1698408015453" src="https://github.com/yuanzhi-zhu/DiffPIR/assets/72423868/12ec14dd-4eef-41ca-badb-530145e25daa">

**Comment by yuanzhi-zhu** [comment]
@xiximelon  That's correct!!
In the paper, **Section 4.1. Implementation Details** reads:
> For inpainting with box-type mask, the mask is 128×128 box region following the approach outlined in [8]; for inpainting with random-type mask, we mask out half of the total pixels at random; for inpainting with prepared mask images, we load the masks used in [38].

**Comment by xiximelon** [comment]
@yuanzhi-zhu Thank you for your detailed and quick response! That’s so nice of you !
